packages/nimble-components/src/table-column/enum-text/group-header-view/index.ts

```diff
@@ -0,0 +1,62 @@
+import { DesignSystem } from '@microsoft/fast-foundation';
+import { observable } from '@microsoft/fast-element';
+import { styles } from '../../text-base/group-header-view/styles';
+import { template } from './template';
+import type { MappingConfigText } from '../../../mapping/text';
+import type { TableColumnEnumColumnConfig } from '../../enum-base';
+import { TableColumnTextGroupHeaderViewBase } from '../../text-base/group-header-view';
+import type { TableFieldValue } from '../../../table/types';
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-table-column-enum-text-group-header-view': TableColumnEnumTextGroupHeaderView;
+    }
+}
+
+/**
+ * A group header view for enum columns
+ */
+export class TableColumnEnumTextGroupHeaderView extends TableColumnTextGroupHeaderViewBase<
+TableFieldValue,
+TableColumnEnumColumnConfig
+> {
+    /** @internal */
+    public span: HTMLSpanElement | null = null;
+
+    /** @internal */
+    @observable
+    public isValidContentAndHasOverflow = false;
+
+    public get mappingToRender(): MappingConfigText | null {
+        const found = this.columnConfig?.mappingConfigs.find(
+            x => x.key === this.groupHeaderValue
+        );
+        return (found as MappingConfigText) ?? null;
+    }
+
+    public override get text(): string {
+        return this.mappingToRender?.label ?? '';
+    }
+
+    public override get placeholder(): string {
+        throw Error('Placeholder not used');
```

Comment:
When grouping by the key value, it is confusing and unhelpful to show potentially multiple, separate groups seemingly grouped by the same placeholder value. Instead, we want to show the actual key value, so you can see why they are separate groups.

Reply:
My concern comes from the fact that this is different behavior than the text column's placeholder. Should the placeholder here have similar behavior to the text column such that it is shown when the column's value is undefined/null rather than being treated as an additional way to configure a default mapping?

Reply:
After we talked about the differences between placeholder text and a default mapping, I came up with a semantic distinction. Placeholder text is for invalid/missing values, whereas a default mapping implies that all values are meaningful. I've added this distinction as usage guidance to the storybook documentation for that parameter. I was also close to just removing the placeholder attribute, but I think there is still enough of a use case to leave it.

Reply:
@mollykreis I'm trying to understand the distinction here. My qs are:

1. If a default mapping is defined then the placeholder is not used right (because every possible value has a 
mapping)? 
2. Should absence of value be treated as a value (does null/undefined map to the default-mapping) or use the placeholder?
3. If a default mapping is not defined (which seems like it is not required for one to be defined) is placeholder used instead?

I think I'm wondering if we should make specifying a default mapping required

Reply:
I'm not sure if I have answers to your questions.

To provide some background, I had raised the concerns I stated earlier in this discussion, particularly these:
- It wasn't clear what the differences were between a default mapping and a placeholder. Which one had precedence over the other, did they behave the same for grouping, etc? As a result, it wasn't clear when someone should use a placeholder vs a default value.
- From what I could tell we weren't treating `placeholder` the same between the mapping columns and our existing text column. I thought that this would be confusing for our clients, and I wanted to make sure it was an explicit decision that everyone agreed on, not just a side-effect of two different developers implementing each column and having different opinions on how things should work.

I discussed this with @m-akinc, but we didn't reach a specific resolution in the discussion, and I don't think I've looked at the code again since it has been updated.

So, to try to answer your questions:
> 1. If a default mapping is defined, then the placeholder is not used right (because every possible value has a
mapping)?

Based on @m-akinc's answer, I don't think that is the case. I believe we would use the placeholder if the value is `null` or `undefined` and maybe in other cases(?). To quote @m-akinc above "Placeholder text is for invalid/missing values".

> 2. Should absence of value be treated as a value (does null/undefined map to the default-mapping) or use the placeholder?

My understanding right now is that the intention is for null/undefined to use the placeholder rather than the default mapping.

> 3. If a default mapping is not defined (which seems like it is not required for one to be defined) is placeholder used instead?

I don't know.


To try to wrap up a very long response, I think we need to take a step back and try to understand what someone would be trying to accomplish by using a default mapping and/or a placeholder for these columns. For example, is the idea of a default mapping to have a short-hand for the multiple values that should map to the same rendered value? Is it because we think clients will be using the mapping column but not know the complete enumeration of possible values that need to be mapped? Is it because some records wouldn't have the field set on them, but they need to be rendered in a specific way?

Reply:
Discussed with the team and decided to remove default-mapping and placeholder from the enum column types. Need to create an issue / resolve if placeholder should be removed from other column types as well (likely not anchor but likely text / numeric formatting columns).